ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -0,0 +1,425 @@
+
+    ed::WorldModel world_model;
+    if (!loadWorldModel(model_name, world_model))
+        return 1;
+
+    tue::filesystem::Path path = argv[1];
+    if (!path.exists())
+    {
+        std::cerr << "Path '" << path << "' does not exist." << std::endl;
+        return 1;
+    }
+
+    std::string entity_id = argv[3];
+
+
+    tue::filesystem::Crawler crawler;
+
+    if (path.isDirectory())
+        crawler.setRootPath(path);
+    else
+        crawler.setRootPath(path.parentPath());
+
+    Updater updater;
+    Fitter fitter;
+
+    std::vector<Snapshot> snapshots;
+    unsigned int i_snapshot = 0;
+
+    while(ros::ok())
+    {
+        if (i_snapshot >= snapshots.size())
+        {
+            bool file_found = true;
+            tue::filesystem::Path filename;
+
+            if (path.isRegularFile() && snapshots.empty())
+                filename = path;
+            else
+            {
+                file_found = false;
+                while (crawler.nextPath(filename))
+                {
+                    if (filename.extension() == ".json")
+                    {
+                        file_found = true;
+                        break;
+                    }
+                }
+            }
+
+            if (file_found)
+            {
+                i_snapshot = snapshots.size();
+                snapshots.push_back(Snapshot());
+                Snapshot& snapshot = snapshots.back();
+                snapshot.world_model = world_model;
+
+                if (!readImage(filename.string(), snapshot.image, snapshot.sensor_pose,
+                               snapshot.world_model, snapshot.area_description))
+                {
+                    std::cerr << "Could not read " << filename << std::endl;
+                    snapshots.pop_back();
+                    continue;
+                }
+                else
+                {
+//                    std::cout << "Successfully loaded " << filename << std::endl;
+                }
+            }
+            else
+            {
+                if (snapshots.empty())
+                    break;
+
+                i_snapshot = snapshots.size() - 1;
+            }
+        }
+
+        Snapshot& snapshot = snapshots[i_snapshot];
+
+        FitterData fitterdata;
+        geo::Pose3D fitted_pose;
+        EstimationInputData result;
+
+        ed::EntityConstPtr e = snapshot.world_model.getEntity(entity_id);
+
+        fitter.configureBeamModel(snapshot.image->getCameraModel());
+        fitter.processSensorData(*snapshot.image, snapshot.sensor_pose, fitterdata);
+
+        bool estimateEntityPose = fitter.estimateEntityPose(fitterdata, snapshot.world_model, entity_id, e->pose(), fitted_pose);
+
+        // show snapshot
+        cv::Mat rgbcanvas = snapshot.image->getRGBImage().clone();
+        cv::imshow("RGB", rgbcanvas);
+
+        // visualise fitting (positive y direction = downwards)
+        int canvas_width = 600;
+        int canvas_height = 600;
```

Comment:
Could this be configurable? Also most images are not square but rectangular.

Reply:
could be. but we would never use that functionality